src/vsphere_cpi/lib/cloud/vsphere/nsxt_provider.rb

```diff
@@ -85,6 +85,11 @@ def to_s
     end
   end
 
+  class NSXTOptimisticUpdateError < StandardError
+    def initialize()
```

Comment:
Doesn't seem like this is necessary. Also it's idiomatic in Ruby to leave off parentheses when calling or defining a function that takes no arguments.